src/Extensions/Abstractions/Activities/ActivityExtensions.cs

```diff
@@ -41,6 +42,27 @@ public static class ActivityExtensions
 		public static Activity MarkAsTransaction(this Activity activity) =>
 			activity.AddBaggage(TransactionMarkerKey, TransactionMarkerValue);
 
+		/// <summary>
+		/// Set result
+		/// </summary>
+		/// <remarks>This property won't be transfered to child activity or via web requests</remarks>
+		public static Activity SetResult(this Activity activity, TimedScopeResult result) =>
```

Comment:
>activity [](start = 49, length = 8)

What happens when activity is null?

Reply:
Activity not nullable here, but if you pass null it will fail with `NullReferenceException`

Reply:
Should we add validation?

---
In reply to: [397166844](https://github.com/microsoft/Omex/pull/173#discussion_r397166844) [](ancestors = 397166844)

Reply:
I don't think it's needed here. 
My current view on null validation is that consumers should use non-nullable reference types or be responsible for not passing null. 
This should reduce the amount of code and simplify validation after moving to non-nullable reference types.